File: lib/common/dto/response_queued_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:netra_flutter/common/models/response.dart';

part 'response_queued_dto.freezed.dart';
part 'response_queued_dto.g.dart';

@freezed
abstract class ResponseQueuedDTO with _$ResponseQueuedDTO {
  const ResponseQueuedDTO._();

  const factory ResponseQueuedDTO({
    required int queueOrder,
  }) = _ResponseQueuedDTO;

  factory ResponseQueuedDTO.fromJson(Map<String, dynamic> json) =>
      _$ResponseQueuedDTOFromJson(json);

  factory ResponseQueuedDTO.fromDataModel(ResponseQueued model) {
    return ResponseQueuedDTO(
      queueOrder: model.queueOrder,
    );
  }

  ResponseQueued toDataModel() {
    return ResponseQueued(
      queueOrder: queueOrder,
    );
  }
}

File: android/src/main/kotlin/com/example/netra_flutter/dto/ResponseQueuedDTO.kt
package com.example.netra_flutter.dto

import com.netra.library.NetraResponse

data class ResponseQueuedDTO(
    val queueOrder: Int,
) {
    companion object {
        fun fromDataModel(response: NetraResponse.ResponseQueued): ResponseQueuedDTO {
            return ResponseQueuedDTO(
                queueOrder = response.queueOrder,
            )
        }
    }

    fun toDataModel(): NetraResponse.ResponseQueued {
        return NetraResponse.ResponseQueued(
            queueOrder = queueOrder,
        )
    }
}
